build failure due to missing dependency on cmi

<!-- Thank you for reporting a bug to Coq! -->

#### Description of the problem
build failure due to missing dependency on cmi
<!-- If you can, it's helpful to provide self-contained example of some code
that reproduces the bug. If not, a link to a larger example is also helpful. -->

#### Coq Version
Coq 8.9.1
camlp5 https://github.com/camlp5/camlp5/pull/37
OCaml 4.08.0
OpenBSD 6.5-current
<!-- You can get this information by running `coqtop -v`. If relevant, please
also include your operating system. -->

#### Fix
``` diff
Index: Makefile.build
--- Makefile.build.orig
+++ Makefile.build
@@ -748,7 +748,7 @@ plugins/%.cmx: plugins/%.ml
 	$(SHOW)'OCAMLOPT  $<'
 	$(HIDE)$(OCAMLOPT) $(COND_OPTFLAGS) $(HACKMLI) $($(@:.cmx=_FORPACK)) -c $<
 
-%.cmx: %.ml
+%.cmx: %.ml %.cmi
 	$(SHOW)'OCAMLOPT  $<'
 	$(HIDE)$(OCAMLOPT) $(COND_OPTFLAGS) $(HACKMLI) -c $<
 
```

Edit: added camlp4, OCaml, OS versions.

I changed the rule because of this error:
```
OCAMLOPT  clib/cObj.ml
File "clib/cObj.ml", line 1:
Error: Could not find the .cmi file for interface clib/cObj.mli.
gmake[1]: *** [Makefile.build:753: clib/cObj.cmx] Error 2
gmake[1]: Leaving directory '/usr/obj/pobj/coq-8.9.1/coq-8.9.1'
gmake: *** [Makefile:166: submake] Error 2
```
With the fix I proposed above the build continued until #10404.

> We also have ml only files, iirc, hence the rule cannot work

ocamlc can generate ``.cmi`` files from ``.ml``, too, can't it?

Oh, I forgot to mention my environment:
Coq 8.9.1
camlp5 https://github.com/camlp5/camlp5/pull/37
OCaml 4.08.0
OpenBSD 6.5-current

--- a/Makefile.build
+++ b/Makefile.build
@@ -262,7 +262,7 @@ endef
 CAMLP5DEPS:=grammar/grammar.cma
 CAMLP5USE=pa_extend.cmo q_MLast.cmo pa_macro.cmo -D$(CAMLVERSION)
 
-PR_O := $(if $(READABLE_ML4),pr_o.cmo,pr_dump.cmo)
+PR_O := $(if $(READABLE_ML4),pr_o.cmo,)
 
 # Main packages linked by Coq.
 SYSMOD:=-package num,str,unix,dynlink,threads

